File: BlazorGameSnakeComponent/wwwroot/BGSnakeCJsInterop.js
var audio_array = [];

window.BGSnakeCJsInterop = {
    Alert: function (a) {
        alert(a);
        return true;
    },
    InitializeSound: function(obj) {
        var index = audio_array.findIndex(x => x === obj["id"]);
        if (index === -1) {
            var audio = new Audio(obj["path"]);
            if (obj["loop"]) {
                audio.loop = true;
            }
            audio_array.push(audio);
        }
        return true;
    },
    ManageSound: function (obj) {
        var audio = audio_array[obj["id"]];
            if (obj["command"] === "play") {
                audio.currentTime = 0;
                audio.play();
            }
            else {
                audio.pause();
            }
        return true;
    }
};

File: BlazorGameSnakeComponent/BGSnakeCJsInterop.cs
using Microsoft.JSInterop;
using System;
using System.Threading.Tasks;

namespace BlazorGameSnakeComponent
{
    public class BGSnakeCJsInterop
    {
        public static IJSRuntime jsRuntime;

        public static ValueTask<bool> Alert(string a)
        {
            return jsRuntime.InvokeAsync<bool>(
                "BGSnakeCJsInterop.Alert", a);
        }


        public static ValueTask<bool> InitializeSound(int id, string path, bool loop)
        {

            return jsRuntime.InvokeAsync<bool>(
                "BGSnakeCJsInterop.InitializeSound", new {id, path, loop });
        }


        public static ValueTask<bool> ManageSound(int id, string command)
        {
            id = id - 1;
            return jsRuntime.InvokeAsync<bool>(
                "BGSnakeCJsInterop.ManageSound", new { id, command});
        }


        
    }
}
